From 02762f03afce3ca1d2d99a11e131526150353524 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Mon, 19 Sep 2011 10:57:56 +0200
Subject: [PATCH] 95iscsi/parse-iscsiroot.sh: fixed iscsi_tcp module loading

modprobing bnx2i provided the sysfs dir, the iscsi_tcp modprobing was
testing for, so iscsi_tcp was never loaded.

https://bugzilla.redhat.com/show_bug.cgi?id=737479
---
 modules.d/95iscsi/parse-iscsiroot.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules.d/95iscsi/parse-iscsiroot.sh b/modules.d/95iscsi/parse-iscsiroot.sh
index 43cb9b9..1c4e72d 100755
--- a/modules.d/95iscsi/parse-iscsiroot.sh
+++ b/modules.d/95iscsi/parse-iscsiroot.sh
@@ -50,7 +50,7 @@ if [ -n "$iscsiroot" ] ; then
     [ -z "$netroot" ] || [ "$netroot" = "iscsi" ] && netroot=iscsi:$iscsiroot
 fi
 
-modprobe bnx2i 2>/dev/null
+[ -e /sys/module/bnx2i ] || modprobe bnx2i 2>/dev/null
 
 # iscsi_firmware does not need argument checking
 if [ -n "$iscsi_firmware" ] ; then
@@ -69,7 +69,7 @@ if [ -z "$iscsi_firmware" ] ; then
 fi
 
 # ISCSI actually supported?
-[ -e /sys/devices/virtual/iscsi_transport ] || modprobe iscsi_tcp || die "iscsiroot requested but kernel/initrd does not support iscsi"
+[ -e /sys/module/iscsi_tcp ] || modprobe iscsi_tcp || die "iscsiroot requested but kernel/initrd does not support iscsi"
 
 # Done, all good!
 rootok=1
